packages/engine/Source/Core/IndexDatatype.js

```diff
@@ -132,8 +132,11 @@ IndexDatatype.createTypedArray = function (
   if (numberOfVertices >= CesiumMath.SIXTY_FOUR_KILOBYTES) {
     return new Uint32Array(indicesLengthOrArray);
   }
+  if (numberOfVertices >= CesiumMath.TWO_FIFTY_SIX_BYTES) {
```

Comment:
I didn't note this when I wrote the issue, but this change should probably apply to `createTypedArrayFromArrayBuffer` as well (the function below this one)